src/vm/comutilnative.cpp

```diff
@@ -1922,6 +1922,21 @@ FCIMPL0(int, GCInterface::GetMaxGeneration)
 }
 FCIMPLEND
 
+FCIMPL0(INT64, GCInterface::GetAllocatedBytesForCurrentThread)
+{
+    FCALL_CONTRACT;
+
+    INT64 currentAllocated = 0;
+    Thread *pThread = GetThread();
+    if (pThread)
+    {
+        alloc_context* ac = pThread->GetAllocContext();
+        currentAllocated = ac->alloc_bytes + ac->alloc_bytes_loh - (ac->alloc_limit - ac->alloc_ptr);
```

Comment:
I see, so that's all the change that was done. I don't think I would do this with this checkin because there was a reason why we were using gen0 on UP because it was faster than getting the alloc_context per thread. of course this was measured many years so it may not affect the perf like it used to be anymore. I would still feel comfortable if I verified that. if it's still noticeably slower I would provide a config for users to specify that they wanna use per thread context on UP when they are ok taking the perf hit (eg, use this config during development time but don't for production).